### PR TITLE
Add linting test

### PR DESCRIPTION
**Pull request by matthewma7**
Add linting tests to the project and fix all linting errors.
See commits for detail.

### CODE REVIEW

**Comment by mwoehlke-kitware** [comment]
> It's OK and known practice to just use js extension for jsx file

...but it's information-losing. With `.jsx`, I know I'm not dealing with basic Javascript. (Also it lets my editor know to highlight JSX rather than plain Javascript.)

File: server/loader.py [hunk 1]
```diff
@@ -23,4 +19,4 @@ def load(info):
                                                      info['serverRoot'])
     info['serverRoot'].api = info['serverRoot'].girder.api
 
-    info['apiRoot'].video = track.TrackResource()
\ No newline at end of file
+    info['apiRoot'].video = track.TrackResource()
```
Comment (mwoehlke-kitware) [inline server/loader.py:22]: *Please* tell me this means lint enforces newline at end of tile? 🙏

Comment (matthewma7) [reply server/loader.py:22]: @mwoehlke-kitware, Yes, the linting enforces newline at the end of files. 

Do you know how to run tests? If you don't, you could try with this girder [documentation](https://girder.readthedocs.io/en/latest/development.html#running-the-tests).

**Comment by matthewma7** [comment]
I agree, jsx would be better, but I didn't find a way to pass parameters to eslint with girder's testing infrastructure. I'll keep this question in mind and try to find people to ask about this, if needed, add the support of eslint parameter. 
Meanwhile, could you try updating your editor or something, my editor actually recognize the jsx part.

**Comment by matthewma7** [comment]
@mwoehlke-kitware Could we approve this one?

**Pull request merged**